tests/ui/single_match.rs

```diff
@@ -81,6 +81,49 @@ fn single_match_know_enum() {
     }
 }
 
+fn issue_173() {
```

Comment:
I'd like some tests that exercise the `&` code

also perhaps we can name this `fn suggest_if` and mention the issue in a comment